lib/drivers/i-sens/careSens.js

```diff
@@ -172,13 +327,142 @@ class CareSens {
       CareSens.verifyChecksum(bytes, response.crc);
     }
 
-    return response;
+    const err = common.getName(ERROR, response.command);
+    if (err !== 'unknown') {
+      throw new Error(err);
+    } else {
+      return response;
+    }
+  }
+
+  static extractPacket(bytes) {
+    const packet = {
+      bytes,
+      valid: false,
+      packet_len: 0,
+      payload: null,
+    };
+
+    const packetLength = bytes.length;
+    packet.packet_len = packetLength;
+    packet.valid = true;
+
+    return packet;
+  }
+
+  static packetHandler(buffer) {
+    if (buffer.len() < 1) { // only empty buffer is no valid packet
+      return false;
+    }
+
+    const packet = CareSens.extractPacket(buffer.bytes());
+    if (packet.packet_len !== 0) {
+      // cleanup the buffer data
+      buffer.discard(packet.packet_len);
+    }
+
+    if (packet.valid) {
+      return packet;
+    }
```

Comment:
It seems that `CareSens.extractPacket()` will _always_ return a packet with the `.valid` set to `true`. If that's the case, then we expect this logic branch to _always_ be run, correct?

Reply:
Yes! I guess that's what happens if you copy & paste a function from another driver and then end up changing it so much that it's almost not necessary anymore.